### ISSUE TITLE
CODI annotation turns separately trained failing regimes into a pathway-collapse test

### ISSUE THREAD

**Issue by microprediction** [microprediction/latentreasoning#114]
## Reviewed source

Current default-branch commit: `88d1db6b1c014f641f67ec4b84ec1f51e8f0c43b`.

Affected annotation: [`bibliography.html` lines 761–765](https://github.com/microprediction/latentreasoning/blob/88d1db6b1c014f641f67ec4b84ec1f51e8f0c43b/bibliography.html#L761-L765):

> On longer hop-lengths CODI does not execute a full latent rollout, instead using a partial pathway concentrated on late intermediates that fuses with the final input position, and that pathway collapses under regime shift.

Primary source: Liang and Pan, [*Do Latent-CoT Models Think Step-by-Step? A Mechanistic Study on Sequential Reasoning Tasks*, arXiv:2602.00449v1](https://arxiv.org/html/2602.00449).

I searched open and closed issues for the paper title and arXiv identifier and found no duplicate.

## “Regime shift” is not an out-of-distribution intervention on a fixed learned pathway

The paper studies purpose-trained CODI students on one synthetic modular-polynomial recurrence,

```
s_t = s_(t-1) x_t + 1 (mod m).
```

The default model is a 3-layer, 2-head transformer with six latent tokens and a 256-dimensional projection, trained for 1,000 epochs on 2,500 examples per sequence length. The long-hop result is specifically for this family, especially composite modulus `m=50`: at four or more hops, probes/logit lens recover mainly the last one or two intermediates, and activation patching is consistent with late fusion of that state with the final input.

The prime/composite comparison is not a distribution shift applied to the same trained model. The released repository provides separate checkpoints and training scripts for `mod=50` and `mod=41`; the checkpoint paths themselves are keyed by modulus. Changing `m` changes the task algebra and its label/token space, so models are trained for each setting rather than taking a fixed composite-trained model and evaluating it out of distribution.

Repository evidence: [official README, separate modulus checkpoints and training commands](https://github.com/jialiang19/latent-cot-thinking#reproducing-the-results).

## The pathway is absent where the model largely fails the task

For prime moduli, the paper reports both loss of late-intermediate decodability **and sharp task failure**:

| modulus | type | CODI accuracy |
|---:|---|---:|
| 41 | prime | 8.67% |
| 43 | prime | 8.59% |
| 47 | prime | 8.31% |
| 42 | composite | 66.40% |
| 45 | composite | 67.05% |
| 48 | composite | 91.07% |

See [§4.4 and Table 1](https://arxiv.org/html/2602.00449#S4.SS4).

The paper’s probe analysis is also performed on activations from examples the model answers correctly. When overall accuracy is around 8%, disappearance of a decodable late-intermediate signature in a separately trained, mostly unsuccessful model is not evidence that a previously functioning pathway “collapsed under regime shift.” It shows that the separately trained prime-modulus models generally failed to learn the task and did not exhibit the composite-task signature.

The algebraic explanation is useful and exact: multiplication by a non-unit modulo a composite `m` is many-to-one, whereas nonzero multiplication modulo a prime is invertible. This makes the composite task more suffix-compressible. But the theorem characterizes the task; it does not establish a robustness failure of a fixed learned mechanism under deployment shift.

## Scope of the long-hop claim

“CODI does not execute a full latent rollout” is also too broad without its denominator. The paper tests small GPT-2-style students trained from scratch on this synthetic recurrence. Its own conclusion proposes testing whether the effect persists across latent-CoT objectives, architectures, and naturalistic data. The result is evidence about **the tested CODI setup on a compressible modular task**, not CODI models generally.

## Citation metadata

The authors are Jia Liang and Liangming Pan, so the citation should be:

> Liang, J., and Pan, L.

The current entry gives `Liang, Y., and Pan, X.`.

## Consequence

The current wording turns a between-training-setting task comparison into an apparent OOD robustness result and obscures that the “collapsed” condition has near-chance/low accuracy. Readers may infer that an otherwise competent CODI model loses its mechanism when inputs shift, which was not tested.

## Suggested correction

> Liang, J., and Pan, L. (2026). *Do Latent-CoT Models Think Step-by-Step? A Mechanistic Study on Sequential Reasoning Tasks*. arXiv:2602.00449. In separately trained 3-layer CODI students on a synthetic modular-polynomial task, composite-modulus runs at long hop counts expose mainly the last one or two intermediates, which fuse with the final input at the answer position. Separately trained prime-modulus models achieve only about 8–9% accuracy and do not show that signature. This supports a task-compressibility hypothesis; it is not an out-of-distribution collapse test on a fixed competent model.

Relevant sources:

- [§4.3, long-hop pathway](https://arxiv.org/html/2602.00449#S4.SS3)
- [§4.4, modulus comparison and accuracies](https://arxiv.org/html/2602.00449#S4.SS4)
- [Appendix E, model/training configuration](https://arxiv.org/html/2602.00449#A5)
- [Official repository](https://github.com/jialiang19/latent-cot-thinking)

**Comment by microprediction** [comment]
Partly accepted in 6dac137. 'Collapse under regime shifts' is the paper's own phrase from its abstract, so the entry wasn't inventing it. But Table 1 compares separately trained students, and the README ships separate checkpoints for mod=50 and mod=41. The prime-modulus students score 8.67/8.59/8.31%. The annotation now says that, and fixes the authors to Liang, J. and Pan, L. One correction to the issue: composite accuracy isn't uniformly high. m=49 gets 19.97% and m=44 gets 34.19%, so I haven't used the 66–91% range.
